java-checks/src/main/java/org/sonar/java/checks/security/FilePermissionsCheck.java

```diff
@@ -0,0 +1,121 @@
+  private static final String ISSUE_MESSAGE = "Make sure this permission is safe.";
+  private static final Set<String> POSIX_OTHER_PERMISSIONS = new HashSet<>(Arrays.asList("OTHERS_READ", "OTHERS_WRITE", "OTHERS_EXECUTE"));
+  private static final MethodMatcher POSIX_FILE_PERMISSIONS_FROM_STRING = MethodMatcher.create()
+    .name("fromString")
+    .typeDefinition("java.nio.file.attribute.PosixFilePermissions")
+    .parameters(JAVA_LANG_STRING);
+
+  private static final MethodMatcher RUNTIME_EXEC = MethodMatcher.create()
+    .name("exec")
+    .typeDefinition("java.lang.Runtime")
+    .withAnyParameters();
+
+  // 'other' group not being 0
+  private static final Pattern CHMOD_OCTAL_PATTERN = Pattern.compile("[0-7]{2}[1-7]");
+  // simplification of all the possible combinations of adding perms to 'other'
+  private static final Pattern SIMPLIFIED_CHMOD_OTHER_PATTERN = Pattern.compile("o[+=](r?w?x?)+");
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Arrays.asList(Tree.Kind.IDENTIFIER, Tree.Kind.METHOD_INVOCATION);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (tree.is(Tree.Kind.IDENTIFIER)) {
+      check((IdentifierTree) tree);
+    } else {
+      check((MethodInvocationTree) tree);
+    }
+  }
+
+  private void check(IdentifierTree identifier) {
+    if (isPosixPermission(identifier) && !isBeingRemoved(identifier)) {
+      reportIssue(identifier, ISSUE_MESSAGE);
+    }
+  }
+
+  private static boolean isPosixPermission(IdentifierTree identifier) {
+    return POSIX_OTHER_PERMISSIONS.contains(identifier.name())
+      && identifier.symbolType().isSubtypeOf("java.nio.file.attribute.PosixFilePermission");
+  }
+
+  private static boolean isBeingRemoved(IdentifierTree identifier) {
+    Tree parent = identifier.parent();
+    while (parent != null) {
+      // Whatever the owner of "remove", we assume the property is dropped if calling a method "remove"
+      // (implemented by all classes extending Collection)
+      if (parent.is(Tree.Kind.METHOD_INVOCATION) && ((MethodInvocationTree) parent).symbol().name().contains("remove")) {
+        return true;
+      }
+      parent = parent.parent();
+    }
+    return false;
+  }
+
+  private void check(MethodInvocationTree mit) {
+    if (POSIX_FILE_PERMISSIONS_FROM_STRING.matches(mit)) {
+      ExpressionTree arg0 = mit.arguments().get(0);
+      if (sensitivePermissionsAsString(arg0)) {
+        reportIssue(arg0, ISSUE_MESSAGE);
+      }
+    } else if (RUNTIME_EXEC.matches(mit)) {
+      ExpressionTree arg0 = mit.arguments().get(0);
+      if (arg0.symbolType().is(JAVA_LANG_STRING) && sensitiveChmodCommand(arg0)) {
+        reportIssue(arg0, ISSUE_MESSAGE);
+      }
+    }
+  }
+
+  private static boolean sensitivePermissionsAsString(ExpressionTree arg0) {
+    return arg0.asConstant(String.class)
+      .filter(chmod -> chmod.length() == 9)
+      .filter(chmod -> !chmod.endsWith("---"))
```

Comment:
Nice and easy way to test: `chmod.endsWith("---")` (you see, from time to time I could agree...:wink:)